### PR TITLE
constraints sample: better way to access remote candidate object

### PR DESCRIPTION
**Pull request by fippo**
**Description**
the current code is rather clumsy. Why search all objects for one that matches id if you can just access it directly.

cc @jan-ivar

### CODE REVIEW

**Comment by KaptenJansson** [comment]
Maybe we couldn't before due to differences in getStats?

**Review by KaptenJansson — APPROVED** [review]
LGTM

**Comment by fippo** [comment]
very legacy. We couldn't do it as easy in the old chrome stats model

File: src/content/peerconnection/constraints/js/main.js [hunk 1]
```diff
@@ -249,13 +249,7 @@ setInterval(function() {
...
       });
       if (activeCandidatePair && activeCandidatePair.remoteCandidateId) {
-        Object.keys(results).forEach(function(result) {
-          var report = results[result];
-          if (report.type === 'remotecandidate' &&
-              report.id === activeCandidatePair.remoteCandidateId) {
-            remoteCandidate = report;
-          }
-        });
+        remoteCandidate = results[activeCandidatePair.remoteCandidateId];
```
Comment (jan-ivar) [inline src/content/peerconnection/constraints/js/main.js:252]: Spec getStats is maplike now, so this needs to be:

```
remoteCandidate = results.get(activeCandidatePair.remoteCandidateId));
```

**Review by jan-ivar — COMMENTED** [review]
Yay, but it should be maplike.

**Comment by fippo** [comment]
small changes, I am not ready for maplike ;-)

**Comment by jan-ivar** [comment]
@fippo Why not? Maplike was polyfilled back in https://github.com/webrtc/adapter/pull/277.

https://jsfiddle.net/sc3dh9wr/ appears to work in Chrome too.

Here's my maplike cheatsheet if it helps inspire: https://jsfiddle.net/4ea0nj0s/

**Comment by fippo** [comment]
@jan-ivar i'd like to have codemod for that

**Pull request merged**